sdks/go/pkg/beam/core/runtime/graphx/serialize.go

```diff
@@ -257,22 +257,22 @@ func decodeFn(u *v1.Fn) (*graph.Fn, error) {
 	if u.Fn != nil {
 		fn, err := decodeUserFn(u.Fn)
 		if err != nil {
-			return nil, fmt.Errorf("bad userfn: %v", err)
+			return nil, fmt.Errorf("failed to decode function proto %v, bad userfn: %v", u, err)
 		}
 		fx, err := funcx.New(reflectx.MakeFunc(fn))
 		if err != nil {
-			return nil, fmt.Errorf("bad userfn: %v", err)
+			return nil, fmt.Errorf("failed to decode function proto %v, bad userfn: %v", u, err)
 		}
 		return &graph.Fn{Fn: fx}, nil
 	}
 
 	t, err := decodeType(u.Type)
 	if err != nil {
-		return nil, fmt.Errorf("bad type: %v", err)
+		return nil, fmt.Errorf("failed to decode function proto %v, bad type: %v", u, err)
 	}
 	fn, err := reflectx.UnmarshalJSON(t, u.Opt)
 	if err != nil {
-		return nil, fmt.Errorf("bad struct encoding: %v", err)
+		return nil, fmt.Errorf("failed to decode function proto %v, bad struct encoding: %v", u, err)
```

Comment:
("failed to decode structural DoFn %v, bad userfn: %v", u, err)

Reply:
Is this a copy/paste error, or did you want "bad struct encoding" changed to "bad userfn"?

Done for the change from "function proto" -> "structural DoFn".

Reply:
The bad userfn was copypasta, sorry about that. Thanks!